chore(semantic-release): update conventional-changelog

josedialpad opened the pull request:
# Update conventional-changelog to make the scope human-readable in the release notes

<!--- Feel free to remove any unused sections -->

## :hammer_and_wrench: Type Of Change

<!--- Tick or place an `x` in all of the checkboxes that apply -->

- [ ] Fix
- [x] Feature
- [ ] Refactoring
- [ ] Documentation

## :book: Description

Unfortunately, this configuration isn't supported out-of-the-box by release-notes-generator (the plugin that is used by semantic-release to generate release notes) so there's no way to format the release notes. So the approach used was to fork the [conventional-changelog-angular package](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular) (which is the convention we use) and add on top of it the [formatting of the scope part of the commit message from `kebab-case` to `Title Case`](https://github.com/josedialpad/conventional-changelog-angular/commit/85f70300306811db413f0d68d6ab2f3c9bab6e9a).

- Project example: https://github.com/josedialpad/dialtone-vue
- Formatted release notes in Github releases: https://github.com/josedialpad/dialtone-vue/releases/tag/v2.30.0
- Changelog: https://github.com/josedialpad/dialtone-vue/blob/staging/CHANGELOG.md

This PR is intended to show the approach used and the result. If it gets a green light, we need to make the fork under the @dialpad org (in github) and publish the package in npm under the @dialpad organization. For testing purposes I used my account in the fork and in the published package in npm.

Conventional changelog license: [ISC License](https://github.com/conventional-changelog/conventional-changelog/blob/master/LICENSE.md) ✅ 

## :bulb: Context

<!--- Describe the purpose of the changes -->
<!--- Why did we make these changes? -->
<!--- What problem(s) do they solve? -->

## :pencil: Checklist

<!--- Tick or place an `x` in all of the checkboxes that apply -->
<!--- Remove checkboxes that do not apply -->

- [x] I have reviewed my changes
- [ ] I have added tests
- [ ] I have added all relevant documentation
- [ ] I have validated components with a screen reader
- [ ] I have validated components keyboard navigation
- [ ] I have considered the performance impact of my change
- [ ] I have checked that my change did not significantly increase bundle size
- [ ] I am exporting any new components or constants in the index.js in the component directory
- [ ] I am exporting any new components or constants in the index.js in the root

## :crystal_ball: Next Steps

- Fork [conventional-changelog-angular package](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular) under a @dialpad repository on Github.
- Apply scope formatting in the fork (https://github.com/josedialpad/conventional-changelog-angular/commit/85f70300306811db413f0d68d6ab2f3c9bab6e9a)
- Publish the package under the @dialpad org in the NPM registry.
- Update this PR with the package.
- Enforce adding the scope in commit messages with `kebab-case` and comma separated when is necessary.

## :camera: Screenshots / GIFs

<!--- Mandatory for any UI work -->
<!--- Link any screenshots / GIFs below -->

## :link: Sources

- https://github.com/semantic-release/release-notes-generator#options

josedialpad commented on .github/COMMIT_CONVENTION.md:
````diff
@@ -40,12 +40,21 @@ Must be one of the following:
+In case of multiple scopes, separate them with comma.
+
+```
+fix(combobox, combobox-with-popover): fix keyboard navigation 
````
> Do you think this will be an issue with messages like this? It's especially a problem with recipe components which usually have longer more descriptive names.

Yeah, 72 chars are not enough in cases where the scope is a recipe and the summary is a bit large. So I increased the limit in commitlint to 120. https://github.com/dialpad/dialtone-vue/pull/537/commits/e5482d3cd40b5560c310cc6c2cd8148773622161

> What happens if they pass in a component name that doesn't exist in our system? Does this matter for our release notes?

No. There's no validation for the scope of the commit message, just formatting the scope for the release notes from [kebab-case to Title Case](https://github.com/dialpad/conventional-changelog-angular/commit/19492e83fbef2a160532978bd596d7ced486a239).

braddialpad replied:
> Yeah, 72 chars are not enough in cases where the scope is a recipe and the summary is a bit large. So I increased the limit in commitlint to 120. [e5482d3](https://github.com/dialpad/dialtone-vue/commit/e5482d3cd40b5560c310cc6c2cd8148773622161)

oh nice, yeah that's better

josedialpad commented:
@braddialpad I created the [conventional-changelog-angular repo](https://github.com/dialpad/conventional-changelog-angular) forked from https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular. The only needed step is to make a release in npm. Added the [release GHA](https://github.com/dialpad/conventional-changelog-angular/blob/staging/.github/workflows/deploy.yml) so setting up the `DIALTONE_NPM_TOKEN` token in that repo should be enough to publish it.

braddialpad commented:
> @braddialpad I created the [conventional-changelog-angular repo](https://github.com/dialpad/conventional-changelog-angular) forked from https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular. The only needed step is to make a release in npm. Added the [release GHA](https://github.com/dialpad/conventional-changelog-angular/blob/staging/.github/workflows/deploy.yml) so setting up the `DIALTONE_NPM_TOKEN` token in that repo should be enough to publish it.

I don't have permissions to this repo can you add me?

josedialpad commented:
> > @braddialpad I created the [conventional-changelog-angular repo](https://github.com/dialpad/conventional-changelog-angular) forked from https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular. The only needed step is to make a release in npm. Added the [release GHA](https://github.com/dialpad/conventional-changelog-angular/blob/staging/.github/workflows/deploy.yml) so setting up the `DIALTONE_NPM_TOKEN` token in that repo should be enough to publish it.
> 
> I don't have permissions to this repo can you add me?

Oh sorry, added with an admin role.

braddialpad commented:
Works now thanks. Lets add a comment at the top of the readme on this forked repo saying why we have done this and what we have changed.

braddialpad commented:
DIALTONE_NPM_TOKEN has been added.

josedialpad commented:
> Works now thanks. Lets add a comment at the top of the readme on this forked repo saying why we have done this and what we have changed.

Good call, added https://github.com/dialpad/conventional-changelog-angular#features-on-top-of-conventional-changelog-angular-v5013
